app/src/components/search/filters/selectFilter.tsx

```diff
@@ -0,0 +1,183 @@
+import { chakra, ChakraComponent } from "@chakra-ui/react";
+import {
+  Box,
+  Button,
+  RadioGroup,
+  Radio,
+  Flex,
+} from "@nypl/design-system-react-components";
+import React, {
+  forwardRef,
+  MutableRefObject,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import FilterAccordion from "./filterAccordion";
+
+export type FilterOption = {
+  name: string;
+  count: number;
+};
+
+export type FilterCategory = {
+  name: string;
+  options: FilterOption[];
+};
+
+export interface SelectFilterProps {
+  filter: FilterCategory;
+}
+
+const SelectFilterComponent = forwardRef<
+  HTMLHeadingElement,
+  React.PropsWithChildren<SelectFilterProps>
+>((props, headingRef) => {
+  const { filter, ...rest } = props;
+  const [userClickedOutside, setUserClickedOutside] = useState<boolean>(false);
+  const [timeoutId, setTimeoutId] = useState<NodeJS.Timeout | null>(null);
+
+  // Create a ref to hold a reference to the accordion button, enabling us
+  // to programmatically focus it.
+  const accordionButtonRef: React.RefObject<HTMLDivElement> =
+    useRef<HTMLDivElement>(null);
+  const containerRef: React.RefObject<HTMLDivElement> =
+    useRef<HTMLDivElement>(null);
+
+  // Tells the accordion to close if open when user clicks outside of the container
+  const handleClickOutside = (e) => {
+    if (e.type === "mousedown") {
+      const multiSelect = containerRef.current;
+      if (multiSelect && !multiSelect.contains(e.target)) {
```

Comment:
Nit: assign the conditional to a variable so you call `setUserClickedOutside(variableName)` once.